lib/files.go

```diff
@@ -0,0 +1,194 @@
+package lib
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// AppendIfMissing appends the content string to the given file
+func AppendIfMissing(filePath string, content string) error {
+	fileToModify, err := os.OpenFile(filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	Check(err)
+
+	fileContents, err := ioutil.ReadFile(fileToModify.Name())
+	Check(err)
+
+	if !strings.Contains(string(fileContents), content) {
+		_, err = fileToModify.Write([]byte(content))
+	}
+
+	fileToModify.Close()
+
+	return err
+}
+
+// RenameDir renames a directory while handling cross-link issues
+// credit https://github.com/rawlingsj/jx/blob/master/pkg/util/files.go
+func RenameDir(src string, dst string, force bool) (err error) {
+	err = CopyDir(src, dst, force)
+	if err != nil {
+		return fmt.Errorf("failed to copy source dir %s to %s: %s", src, dst, err)
+	}
+	err = os.RemoveAll(src)
+	if err != nil {
+		return fmt.Errorf("failed to cleanup source dir %s: %s", src, err)
+	}
+	return nil
+}
+
+// RenameFile renames a file while handling cross-link issues
+// credit https://github.com/rawlingsj/jx/blob/master/pkg/util/files.go
+func RenameFile(src string, dst string) (err error) {
```

Comment:
Similar to my RenameDir comment: This has slightly different semantics to rename as it could leave a partial file at the destination.  If you want the same semantics without cross-device link issues, you'll want, to do a copy to a temptfile then a rename.

Reply:
This function declares a named returned, but then directly returns an error in each case, consider removing the named return for clarity.